crates/shared/src/signature_validator/simulation.rs

```diff
@@ -66,13 +66,22 @@ impl Validator {
         // change), the order's validity can be directly determined by whether
         // the signature matches the expected hash of the order data, checked
         // with isValidSignature method called on the owner's contract
-        let contract = ERC1271SignatureValidator::at(&self.web3, check.signer);
+        let contract = ERC1271SignatureValidator::Instance::new(
+            check.signer.into_alloy(),
+            self.web3.alloy.clone(),
```

Comment:
We don't need a long lived contract instance here so passing just a reference of the alloy provider is sufficient. For that you'll have to use `ERC1271SignatureValidator::ERC1271SignatureValidator::new`, though. But given that it's only used in one place here you can just import the inner `ERC1271SignatureValidator` directly in the `use` block.